### PR TITLE
Fix dockerfile parsing in sendmail plugin

### PR DESCRIPTION
**Pull request by chmeliik**
CLOUDBLD-8266

Parse the dockerfile from any build dir, not the source dir.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/exit_sendmail.py [hunk 1]
```diff
@@ -153,7 +152,9 @@ def _should_send(self, success, manual_canceled):
     def _get_image_name_and_repos(self):
 
         repos = []
-        dockerfile = df_parser(self.workflow.df_path, workflow=self.workflow)
+        dockerfile = self.workflow.build_dir.any_platform.dockerfile_with_parent_env(
```
Comment (hjmodi) [inline atomic_reactor/plugins/exit_sendmail.py:155]: This will not work without initializing the build dirs

**Comment by hjmodi** [comment]
I am thinking we should remove `df_path` entirely. While testing flatpak builds I realized that `df_path` doesn't get set since flatpak sources don't have a Dockerfile in source. Instead, we can set the platform independent data in `workflow.data` and use that. WDYT?

File: atomic_reactor/plugins/exit_sendmail.py (hunk 1 above)
Comment (chmeliik) [reply atomic_reactor/plugins/exit_sendmail.py:155]: Oh, good catch! That's why we need consistency /shakes_fist

**Comment by chmeliik** [comment]
> I am thinking we should remove `df_path` entirely. While testing flatpak builds I realized that `df_path` doesn't get set since flatpak sources don't have a Dockerfile in source. Instead, we can set the platform independent data in `workflow.data` and use that. WDYT?

Definitely remove, it's on the list of things to be removed at some point

What do you mean by platform independent data?

**Comment by hjmodi** [comment]
> > I am thinking we should remove `df_path` entirely. While testing flatpak builds I realized that `df_path` doesn't get set since flatpak sources don't have a Dockerfile in source. Instead, we can set the platform independent data in `workflow.data` and use that. WDYT?
> 
> Definitely remove, it's on the list of things to be removed at some point
> 
> What do you mean by platform independent data?

I am thinking that the Dockerfile might be modified during prebuild, with platform-specific data but my guess is that some things like labels may remain the same for all platforms. We can capture that data if needed instead of trying to retrieve it from `dockerfile` in the exit plugins.

**Comment by chmeliik** [comment]
> I am thinking that the Dockerfile might be modified during prebuild, with platform-specific data but my guess is that some things like labels may remain the same for all platforms. We can capture that data if needed instead of trying to retrieve it from `dockerfile` in the exit plugins.

Hmm, might be worth looking into, but it won't be too simple. We would have to define the data that is platform-independent, and probably have a new plugin / whatever to set this data at the end of the prebuild phase, or at some point in postbuild.

Labels are not platform-indepedent, there is at least one (architecture) that's different between platforms. The name-version-release labels should be independent, and NVR is one of the main use cases of parsing the Dockerfile, so that could be an option. But if we were to do that, we should make it _the_ way to get NVR instead of letting all prebuild plugins do it however they like

TL;DR - for now, the easier solution seems to be letting exit plugins use the dockerfile :)

**Review by rcerven — APPROVED** [review]

**Review by tkdchen — APPROVED** [review]

**Comment by hjmodi** [comment]
@chmeliik do you need to waive coverage?

**Comment by chmeliik** [comment]
> @chmeliik do you need to waive coverage?

@hjmodi I checked the coveralls report and the 2 newly missed lines are in the `util.df_parser()` method which should be deleted when it's no longer needed (which should be soon, this might have been its last use outside of unit tests)

**Review by hjmodi — APPROVED** [review]

**Comment by hjmodi** [comment]
LGTM
